Fedora rawhide check

pemensik opened the pull request:
This is attempt to reduce number of packages, which need to be updated after new release is branched from current rawhide.  This were initiated by broken mock during branching, because updates were needed together in two separate components:

- https://bodhi.fedoraproject.org/updates/FEDORA-2024-1a79132182
- https://bodhi.fedoraproject.org/updates/FEDORA-2024-be19a8b7dd

I think that is in fact not necessary, because mock-core-configs uses releasever only to compute key path one higher and one lower than current rawhide. If that is provided directly by this package, then mock-core-config can use symlink provided and change only when they want to add new release into collection. This package update would fix also mock rawhide.

pemensik committed:
Check and update Fedora Rawhide release version

Requires fedora-repos-rawhide installed. Tests whether current Fedora
rawhide points to correct release key. Creates also rawhide+1 and
rawhide-1 links to preceding releases. Might be used by
mock-core-configs package directly.

pemensik committed:
Add new rawhide relative links

Upcoming rawhide release key and previous release key is often required
by various tools. Prepare symlinks, which can be updated by script on
next Fedora branching.

pemensik commented:
It is also related to bug https://bugzilla.redhat.com/show_bug.cgi?id=2305309

xsuchy commented:
This is already handled by https://github.com/rpm-software-management/distribution-gpg-keys/blob/main/.github/workflows/fedora-repos.yml

And check the dates. The new GPG key was added here very shortly after the key was created by relengs.

pemensik commented:
@xsuchy That workflow checks only addition of new GPG key.

But does not trigger any action when rawhide is switched to new release. Even though it were updated quickly, it still has broken mockbuild on rawhide branch, because the update was not done together with *mock-core-configs* (and _fedora-repos_). Ideally we should have just one source of truth, which change at single place would be enough to switch to N+1 release in all other tools. The less packages, where that change needs to happen, the better. This is one step to remove the need to update releasever in mock-core-configs. Because it uses the key from this package anyway and that change needs to happen here first.

It does not count when key is added, but when the update hits the stable. That unfortunately were not as quick and multiple people complained on Friday at devel matrix channel. That could be avoided if change of mock-core-config did not have to match change in distribution-gpg-keys.

pemensik committed:
Check rawhide key points to correct release key

Obtain rawhide_release from spec file and check our rawhide key points
to matching release.

pemensik commented:
These symlinks should avoid the need to change *mock-core-configs/etc/mock/templates/fedora-rawhide.tpl* and *rawhide-eln.tpl* in https://github.com/rpm-software-management/mock/commit/1d000d9b907bd1a64aa5cae9a05b0a6de3aa159e#diff-6d85f049d29a9ebeda1e02d48fb369982d6410ba63c64306001e1f41f8a3c4dd. Which needs to match change here and in current way needs to part of the same update, probably using side tags.

pemensik commented:
That is, these changes are needed because rawhide_gpg_keys macro creates rawhide+1 and rawhide-1 key links. If this package provides these links, the only this package is responsible for the change and releasever adjusting from mock-core-configs can be removed. Replaced by just something like:

```
{%- macro rawhide_gpg_keys() -%}
file:///usr/share/distribution-gpg-keys/fedora/RPM-GPG-KEY-fedora-rawhide-primary
file:///usr/share/distribution-gpg-keys/fedora/RPM-GPG-KEY-fedora-rawhide-1-primary
file:///usr/share/distribution-gpg-keys/fedora/RPM-GPG-KEY-fedora-rawhide-2-primary
{%- endmacro %}
```

But wait. I have not noticed eln uses different values than rawhide. rawhide uses ``rawhide+0, rawhide-1, rawhide+1``, but *eln* uses ``rawhide+0, rawhide-1, rawhide-2``. So perhaps rawhide-2 must be maintained too here too.

pemensik committed:
Add also rawhide-2 for use in eln

fedora-eln template uses rawhide+0, rawhide-1 and rawhide-2 keys. Unlike
rawhide +1 and -1. Add additional managed link too.

pemensik committed:
Switch to git fetching fedora-repos for rawhide release

GitHub worflow already contains similar check. Use similar also for
checking rawhide release here. It should work on any distribution having
git and awk installed, for example ubuntu image.

pemensik commented:
This is related to problem https://github.com/rpm-software-management/mock/issues/1338, which needs similar workarounds also in mkosi. Having this at key provider might provide decent workaround.
